homeassistant/components/workday/binary_sensor.py

```diff
@@ -103,7 +103,20 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     # Remove holidays
     try:
         for date in remove_holidays:
-            obj_holidays.pop(date)
+            # is this formatted as a date?
+            if dt.parse_date(date):
+                # remove holiday by date
+                removed = obj_holidays.pop(date)
+                _LOGGER.debug("Removed %s", date)
+            else:
+                try:
+                    # remove holiday by name
+                    _LOGGER.debug("Treating '%s' as named holiday", date)
+                    removed = obj_holidays.pop_named(date)
+                    for holiday in removed:
+                        _LOGGER.debug("Removed %s by name '%s'", holiday, date)
+                except KeyError as unmatched:
+                    _LOGGER.warning("No holiday found matching %s", unmatched)
```

Comment:
Can we just move the new try up one level to cover the unknown date case too: I.e.,

```python
        for date in remove_holidays:
            try:
                # is this formatted as a date?
                if dt.parse_date(date):
                    # remove holiday by date
                    removed = obj_holidays.pop(date)
                    _LOGGER.debug("Removed %s", date)
                else:
                    # remove holiday by name
                    _LOGGER.debug("Treating '%s' as named holiday", date)
                    removed = obj_holidays.pop_named(date)
                    for holiday in removed:
                        _LOGGER.debug("Removed %s by name '%s'", holiday, date)
            except KeyError as unmatched:
                _LOGGER.warning("No holiday found matching %s", unmatched)
```

Reply:
Done.  Thanks @TomBrien.